module/zfs/arc.c

```diff
@@ -3512,22 +3512,44 @@ arc_hdr_realloc_crypt(arc_buf_hdr_t *hdr, boolean_t need_crypt)
 	arc_buf_hdr_t *nhdr;
 	arc_buf_t *buf;
 	kmem_cache_t *ncache, *ocache;
+	unsigned nsize, osize;
 
 	ASSERT(HDR_HAS_L1HDR(hdr));
 	ASSERT3U(!!HDR_PROTECTED(hdr), !=, need_crypt);
 	ASSERT3P(hdr->b_l1hdr.b_state, ==, arc_anon);
 	ASSERT(!multilist_link_active(&hdr->b_l1hdr.b_arc_node));
+	ASSERT(!list_link_active(&hdr->b_l2hdr.b_l2node));
```

Comment:
It isn't obvious why this assertion is true.  I think it's because this is only called on ANON hdr's, which can't be in l2arc?  Maybe add a comment explaining this?

Reply:
That is correct, I can add a comment.